cpp/lib/src/HtmlUtil.cc

```diff
@@ -1019,4 +1019,32 @@ std::string StripHtmlTags(const std::string &text_with_optional_tags, const bool
 }
 
 
+std::string ShortenText(const std::string &html_text, const unsigned max_length) {
+    std::string shortened_text;
+    shortened_text.reserve(html_text.size());
+
+    bool in_tag(false);
+    unsigned length(0);
+
+    static const std::string placeholder("...");
+    unsigned max_length_without_placeholder(max_length - placeholder.length());
+    for (auto ch(html_text.cbegin()); ch != html_text.cend(); ++ch) {
```

Comment:
Any particular reason why you didn't use an implicit for-loop?

Reply:
I tried to find an example for a similar implementation and copied code from HtmlUtil::StripHtmlTags.

Reply:
That is ancient code.  Please use "for (const char ch : html_text)" and replace "*ch" with "ch" everywhere in the loop body.